repose-aggregator/components/filters/cors-filter/src/test/scala/org/openrepose/filters/cors/CorsFilterTest.scala

```diff
@@ -156,18 +158,27 @@ class CorsFilterTest extends FunSpec with BeforeAndAfterEach with Matchers {
           corsFilter.doFilter(servletRequest, servletResponse, filterChain)
 
           servletResponse.getHeader(CorsHttpHeader.ACCESS_CONTROL_ALLOW_METHODS) should not be null
+          servletResponse.getHeaders(CorsHttpHeader.ACCESS_CONTROL_ALLOW_METHODS) should have size 1
         }
 
-        List("X-Auth-Token", "X-Panda, X-Unicorn", "Accept, User-Agent, X-Trans-Id").foreach { requestHeader =>
-          it(s"should have the Access-Control-Allow-Headers header set for request HTTP method $requestMethod and request headers $requestHeader") {
+        List(
+          (List("x-auth-token"), List("x-auth-token")),
+          (List("x-panda, x-unicorn"), List("x-panda", "x-unicorn")),
+          (List("x-cupcake", "x-pineapple"), List("x-cupcake", "x-pineapple")),
+          (List("accept, user-agent, x-trans-id"), List("accept", "user-agent", "x-trans-id")),
+          (List("x-one, x-two", "x-three"), List("x-one", "x-two", "x-three")),
+          (List("x-red", "x-green", "x-blue"), List("x-red", "x-green", "x-blue"))
```

Comment:
The test is making sure that whether the browser sends in the `Access-Control-Request-Headers` header values split or merged or some mess in between, the CorsFilter should include all of the values in the response header `Access-Control-Allow-Headers`.  The 1st, 3rd, and 6th lists are the same.